tests/functional/adapter/test_grant_access_to.py

```diff
@@ -21,21 +22,47 @@
 ) }}
 SELECT 1 as one
 """
+BAD_CONFIG_TABLE_NAME = "bad_view"
+BAD_CONFIG_TABLE = """
+{{ config(
+    materialized='view',
+    grant_access_to=[
+      {'project': 'dbt-test-env-alt', 'dataset': 'NonExistentDataset'},
+    ]
+) }}
+SELECT 1 as one
+"""
 
+BAD_CONFIG_CHILD_TABLE = "SELECT 1 as one FROM {{ref('" + BAD_CONFIG_TABLE_NAME + "')}}"
 
-class GrantAccessToConfig:
+
+class TestAccessGrantSucceeds:
     @pytest.fixture(scope="class")
     def models(self):
-        return {
-            "select_1.sql": SELECT_1,
-            "select_1_table.sql": SELECT_1_TABLE,
-        }
-
+        return {"select_1.sql": SELECT_1, "select_1_table.sql": SELECT_1_TABLE}
 
-class TestAccessGrantSucceeds(GrantAccessToConfig):
     def test_grant_access_succeeds(self, project):
-        #Need to run twice to validate idempotency
+        # Need to run twice to validate idempotency
         results = run_dbt(["run"])
         assert len(results) == 2
         results = run_dbt(["run"])
         assert len(results) == 2
+
+
+class TestAccessGrantFails:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "bad_config_table_child.sql": BAD_CONFIG_CHILD_TABLE,
+            f"{BAD_CONFIG_TABLE_NAME}.sql": BAD_CONFIG_TABLE,
+        }
+
+    def test_grant_access_fails_without_running_child_table(self, project):
+        # Need to run twice to validate idempotency
+        results = run_dbt(["run"], expect_pass=False)
+        assert results[0].status == "error"
+        assert results[1].status == "skipped"
+        assert (
+            results[0].message
+            == "404 GET https://bigquery.googleapis.com/bigquery/v2/projects/dbt-test-env-alt/datasets/NonExistentDataset?prettyPrint=false: Not found: Dataset dbt-test-env-alt:NonExistentDataset"
```

Comment:
Do we specifically want to enforce this entire string? It looks very specific, which suggests it might also be brittle. We might wind up with a failed test in the future that doesn't reflect broken functionality. Can we loosen the constraint to something like:
```
assert results[0].message.startswith("404 GET https://bigquery.googleapis.com/")
```